lib/rules/declaration-property-value-keyword-no-deprecated/index.mjs

```diff
@@ -0,0 +1,157 @@
+	inactivecaption: 'canvas',
+	infobackground: 'canvas',
+	menu: 'canvas',
+	scrollbar: 'canvas',
+	window: 'canvas',
+	activeborder: 'ButtonBorder',
+	inactiveborder: 'ButtonBorder',
+	threeddarkshadow: 'ButtonBorder',
+	threedhighlight: 'ButtonBorder',
+	threedlightshadow: 'ButtonBorder',
+	threedshadow: 'ButtonBorder',
+	windowframe: 'ButtonBorder',
+	captiontext: 'CanvasText',
+	infotext: 'CanvasText',
+	menutext: 'CanvasText',
+	windowtext: 'CanvasText',
+	buttonhighlight: 'ButtonFace',
+	buttonshadow: 'ButtonFace',
+	threedface: 'ButtonFace',
+	inactivecaptiontext: 'GrayText',
+};
+const COLOR_PROPERTIES = [...singleValueColorProperties.values()].map(
+	(value) => /** @type {[string, Record<String, string>]} */ ([value, DEPRECATED_COLORS]),
+);
+const PROPERTY_NAME_TO_KEYWORD = new Map([
+	[
+		'appearance',
+		/** @type {string | string[] | Record<string, string>} */ ({
+			button: 'auto',
+			checkbox: 'auto',
+			listbox: 'auto',
+			menulist: 'auto',
+			meter: 'auto',
+			'progress-bar': 'auto',
+			'push-button': 'auto',
+			radio: 'auto',
+			searchfield: 'auto',
+			'slider-horizontal': 'auto',
+			'square-button': 'auto',
+			textarea: 'auto',
+		}),
+	],
+	[
+		'image-rendering',
+		{
+			optimizeQuality: 'smooth',
+			optimizeSpeed: 'pixelated',
+		},
+	],
+	['overflow', { overlay: 'auto' }],
+	['text-justify', { distribute: 'inter-character' }],
+	['text-orientation', { 'sideways-right': 'sideways' }],
+	['user-select', { element: 'contain' }],
+	['zoom', { reset: '1' }],
+	['text-decoration-line', 'blink'],
+	['box-sizing', 'padding-box'],
+	['image-orientation', 'flip'],
+	['min-height', ['intrinsic', 'min-intrinsic']],
+	['min-width', ['intrinsic', 'min-intrinsic']],
+	['max-height', ['intrinsic', 'min-intrinsic']],
+	['max-width', ['intrinsic', 'min-intrinsic']],
+	['height', ['intrinsic', 'min-intrinsic']],
+	['width', ['intrinsic', 'min-intrinsic']],
+	['word-break', 'break-word'],
+	...COLOR_PROPERTIES,
+]);
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				optional: true,
+				actual: secondaryOptions,
+				possible: {
+					ignoreValues: [isString, isRegExp],
+				},
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkDecls((decl) => {
+			const declarationValue = decl.value;
+
+			if (optionsMatches(secondaryOptions, 'ignoreValues', declarationValue)) return;
+
+			const value = PROPERTY_NAME_TO_KEYWORD.get(decl.prop.toLowerCase());
+
+			if (!value) return;
+
+			const keyword = declarationValue.toLowerCase();
+			let fix;
+
+			if (typeof value === 'string' && value !== keyword) return;
+
+			if (Array.isArray(value) && !value.includes(keyword)) return;
+
+			if (isPlainObject(value)) {
+				const fixedValue = value[keyword];
+
+				if (!fixedValue) return;
+
+				fix = () => {
+					decl.value = fixedValue;
```

Comment:
```suggestion
					decl.value = fixedValue;
```

To fix values with comments, we may need to use the `get/setDeclarationValue` utils and the value parser.

Reply:
I did it initially but it was not necessary.
I have compared the resulting nodes and added regression tests just in case.
see https://github.com/stylelint/stylelint/pull/8223/files#diff-652c44ce4d315110a294f6324f0b75cfda7f1ae30e1c82c9233b7d31b62fff05R33 https://github.com/stylelint/stylelint/pull/8223/files#diff-652c44ce4d315110a294f6324f0b75cfda7f1ae30e1c82c9233b7d31b62fff05R42
cf 9599633

Reply:
The missing `fix: true` mislead me.
Ill fix it.